src/tech/v3/dataset/column_index_structure.clj

```diff
@@ -28,3 +28,62 @@
    (select-from-index index-structure mode selection-spec {:as-index-structure false}))
   ([index-structure mode selection-spec options]
    (col-proto/select-from-index index-structure mode selection-spec options)))
+
+
+(comment 
+  (require '[tech.v3.dataset :as ds])
+  (require '[tech.v3.dataset.column :as col])
+  (require '[criterium.core :as c])
+
+  (def x (col/new-column :x (range 200000)))
+
+  (def xidx (col/index-structure x))
+
+  (c/quick-bench
+  (def x-indices (select-from-index xidx :slice {:from 100000 :to 200000})))
+
+  (def x-indices (select-from-index xidx :slice {:from 100000 :to 200000}))
+
+
+
+  (c/quick-bench
+  (-> xidx
+      (.subMap 100000 true 200000 true)
+      (.values)
+      (->> (reduce into (tech.v3.datatype.ListPersistentVector. [])))))
+  ;; Execution time mean : 23.044314 sec
+
+  (c/quick-bench
+  (-> xidx
+      (.subMap 100000 true 200000 true)
+
+      (.values)
+      (->> (reduce concat []))
+      (tech.v3.datatype.ListPersistentVector.)
+      ))
+  ;; Execution time mean : 2.879188 ms
```

Comment:
This is a fake result.  Trying to access every element in the result will result in a stack overflow exception due to the [way concat works](https://stackoverflow.com/questions/24958907/why-does-reduce-give-a-stackoverflowerror-in-clojure).  Furthermore the result isn't random access any more so any next operation such as selecting some of those indexes will result in the sequence being realized.  For this type of use you need to use `(apply concat ...)` not `(reduce concat ...)

Reply:
And even then you get a stack overflow exception if you try to access every element.  Let's throw this one out - it isn't a viable pathway.